presto-main/src/main/java/com/facebook/presto/cost/FragmentedPlanSourceProvider.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.plan.PlanFragmentId;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.RemoteSourceNode;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+public class FragmentedPlanSourceProvider
+        implements PlanNodeSourceProvider
+{
+    private Map<PlanFragmentId, PlanNode> fragmentPlanNodeMap;
```

Comment:
`planFragments`?

Reply:
Changing to planFragmentsMap.  I don't like when maps are named without the word map (or something that makes it clear it's a mapping).  I always expect them to be a list or a set (or any other collection of just one thing).

Reply:
Heh... I actually dislike when variable names encode their type (unless there's a very good reason for that, such as a conflict with some other variable -- e.g., a when a builder is used, or when having to convert types and you need to hold on to both versions). In this particular case, fragmentPlanNodeMap doesn't even convey what it represents. I would suggest sourcesByFragment, nodesByFragment, sourceNodesByFragment or something like that.